bodhi/server/util.py

```diff
@@ -684,3 +686,101 @@ def sort_severity(value):
     }
 
     return value_map.get(value, 99)
+
+
+def get_critpath_pkgs_from_pdc(branch):
+    """
+    Searches PDC for critical path packages based on the specified branch
+    :param branch: string representing the branch
+    :return: a list of critical path packages
+    """
+    pdc_api_url = '{}/rest_api/v1/component-branches/'.format(
+        config.get('pdc_url', 'https://pdc.fedoraproject.org/').rstrip('/'))
+    query_args = urllib.urlencode({
+        'active': 'true',
+        'critical_path': 'true',
+        'name': branch,
+        'page_size': 100,
+        'type': 'rpm'
+    })
+    pdc_api_url_with_args = '{0}?{1}'.format(pdc_api_url, query_args)
+
+    critpath_pkgs_set = set()
+    while True:
+        pdc_request_json = pdc_api_get(pdc_api_url_with_args)
+
+        for branch in pdc_request_json['results']:
+            critpath_pkgs_set.add(branch['global_component'])
+
+        if pdc_request_json['next']:
+            pdc_api_url_with_args = pdc_request_json['next']
+        else:
+            # There are no more results to iterate through
+            break
+    return list(critpath_pkgs_set)
+
+
+def pagure_api_get(pagure_api_url, timeout=60):
+    """
+    A wrapper to get API resources from Pagure
+    :param pagure_api_url: a string of the URL to query
+    :param timeout: the timeout on the GET request
+    :return: JSON of the API resource(s) or a RuntimeError
+    """
+    base_error_msg = (
+        'Bodhi failed to get a resource from Pagure at the following URL '
+        '"{0}". The status code was "{1}"')
+    rv = requests.get(pagure_api_url, timeout=timeout)
+
+    if rv.status_code == 200:
+        return rv.json()
+    elif rv.status_code == 500:
+        # There will be no JSON with an error message here
+        error_msg = base_error_msg.format(pagure_api_url, rv.status_code)
+        log.exception(error_msg)
```

Comment:
How about ```log.error```? ```log.exception``` will put a traceback in the log even though it's not actually a Bodhi crash.